vagrant/provisioning/roles/elasticsearch-install/tasks/main.yml

```diff
@@ -0,0 +1,54 @@
+---
+    # tasks file for /etc/ansible/roles/elastic
+    - name: Copy the ElasticSearch repo
+      template:
+        src: elasticsearch.repo
+        dest: /etc/yum.repos.d/
+        backup: yes
+    
+    - name: Import GPG key for ElasticSearch   
+      rpm_key:
+        state: present
+        key: https://artifacts.elastic.co/GPG-KEY-elasticsearch      
+
+    - name: Install ElasticSearch
+      yum:
+        name: elasticsearch
+        state: present
+
+    - name: Elasticsearch folders
+      become: yes
+      file:
+        path: "{{ item }}"
+        state: directory
+        group: elasticsearch
+        owner: elasticsearch
+      loop:
+        - "{{ root_folder }}/data/elasticsearch"
+        - "{{ root_folder }}/log/elasticsearch"
+
+    - name: Copying ElasticSearch configuration files
+      template:
+        src: elasticsearch.yml
+        dest: /etc/elasticsearch
+        backup: yes
+    
+    - name: Adding services to auto start for CentOS 6
+      service:
+        name: elasticsearch
+        enabled: yes
+        state: started
+      when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '6'and enable_elasticsearch is defined and enable_elasticsearch == "yes"
+
+    - name: Adding services to auto start for CentOS 7
+      systemd:
+        name: elasticsearch
+        enabled: yes
+        daemon_reload: yes
+        masked: no
+        state: started
+      when: ansible_distribution == 'CentOS' and ansible_distribution_major_version == '7' and enable_elasticsearch is defined and enable_elasticsearch == "yes"
+ 
+    - name: Wait for 20 seconds for elasticsearch to become responsive...
+      wait_for: timeout=20
+      delegate_to: localhost
```

Comment:
When running this role on the AWX server, will this 'localhost' be the AWX server itself, or the remote host?

Reply:
@david-oc-miller, since 'localhost' will be the AWX server itself, i removed 'delegate_to'. Now this part of the task will be running on remote host.
Dimitar told me that 'delegate_to: localhost' was using when you didn't have AWX server.